R/FilterStatesDF.R

```diff
@@ -28,11 +36,11 @@ DFFilterStates <- R6::R6Class( # nolint
     #' @param keys (`character`)\cr
     #'   key columns names
     #'
-    initialize = function(input_dataname, output_dataname, datalabel, varlabels, keys) {
-      super$initialize(input_dataname, output_dataname, datalabel)
+    initialize = function(data, data_reactive, input_dataname, output_dataname, datalabel, varlabels, keys) {
+      super$initialize(data, data_reactive, input_dataname, output_dataname, datalabel)
       private$varlabels <- varlabels
       private$keys <- keys
-
+      self$set_filterable_varnames(colnames(data))
```

Comment:
This seems to work nicely both in a standalone app and also inside a full filterpanel (where filtereddataset overwrites it) - I'm not seeing a corresponding change for the other FilterStates classes though

Reply:
Yes, `"filterable"` is not implemented for MAE. Never was

For MAEFilterStates `srv_add_filter_state` doesn't use any field nor argument which is connected with "filterable".

https://github.com/insightsengineering/teal.slice/blob/5909acbf662b70ba5aa04717efdfdfea63ff6fc6/R/FilterStatesMAE.R#L285

https://github.com/insightsengineering/teal.slice/blob/5909acbf662b70ba5aa04717efdfdfea63ff6fc6/R/FilteredDatasetMAE.R#L158

https://github.com/insightsengineering/teal.slice/blob/5909acbf662b70ba5aa04717efdfdfea63ff6fc6/R/FilterStates-utils.R#L188



P.S. When we start API issues it will be clear what is missing.